service/chain/disk_test.go

```diff
@@ -32,286 +30,407 @@ import (
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		c := chain.FromDisk(db)
+
+		_, err := c.Collections(mocks.GenericHeight)
+		assert.Error(t, err)
+	})
 }
 
 func TestDisk_Guarantees(t *testing.T) {
-	db := populateDB(t)
-	defer db.Close()
-	c := chain.FromDisk(db)
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(mocks.GenericHeight, mocks.GenericHeader.ID())))
+		require.NoError(t, db.Update(operation.IndexPayloadGuarantees(mocks.GenericHeader.ID(), mocks.GenericCollectionIDs(2))))
+		require.NoError(t, db.Update(operation.InsertGuarantee(mocks.GenericCollection(0).ID(), mocks.GenericGuarantee(0))))
+		require.NoError(t, db.Update(operation.InsertGuarantee(mocks.GenericCollection(1).ID(), mocks.GenericGuarantee(1))))
+
+		c := chain.FromDisk(db)
+
+		tt, err := c.Guarantees(mocks.GenericHeight)
+		assert.NoError(t, err)
+		assert.Len(t, tt, 2)
+
+		_, err = c.Guarantees(math.MaxUint64)
+		assert.Error(t, err)
+	})
+
+	t.Run("handles missing entry for indexed height", func(t *testing.T) {
+		t.Parallel()
 
-	tt, err := c.Guarantees(mocks.GenericHeight)
-	assert.NoError(t, err)
-	assert.Len(t, tt, 2)
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
 
-	_, err = c.Guarantees(math.MaxUint64)
-	assert.Error(t, err)
+		require.NoError(t, db.Update(operation.IndexBlockHeight(mocks.GenericHeight, mocks.GenericHeader.ID())))
+
+		c := chain.FromDisk(db)
+
+		_, err := c.Guarantees(mocks.GenericHeight)
+		assert.Error(t, err)
+	})
+
+	t.Run("handles call on non-indexed height", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		c := chain.FromDisk(db)
+
+		_, err := c.Guarantees(mocks.GenericHeight)
+		assert.Error(t, err)
+	})
 }
 
 func TestDisk_Transactions(t *testing.T) {
-	db := populateDB(t)
-	defer db.Close()
-	c := chain.FromDisk(db)
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(mocks.GenericHeight, mocks.GenericHeader.ID())))
+		require.NoError(t, db.Update(operation.IndexPayloadGuarantees(mocks.GenericHeader.ID(), mocks.GenericCollectionIDs(2))))
+		require.NoError(t, db.Update(operation.InsertCollection(mocks.GenericCollection(0))))
+		require.NoError(t, db.Update(operation.InsertTransaction(mocks.GenericTransaction(0).ID(), mocks.GenericTransaction(0))))
+		require.NoError(t, db.Update(operation.InsertTransaction(mocks.GenericTransaction(1).ID(), mocks.GenericTransaction(1))))
+		require.NoError(t, db.Update(operation.InsertCollection(mocks.GenericCollection(1))))
+		require.NoError(t, db.Update(operation.InsertTransaction(mocks.GenericTransaction(2).ID(), mocks.GenericTransaction(2))))
+		require.NoError(t, db.Update(operation.InsertTransaction(mocks.GenericTransaction(3).ID(), mocks.GenericTransaction(3))))
```

Comment:
OK maybe the `err :=` style would be too verbose here. I take that back. Makes sense to shorten it like this.

Reply:
Ah, I just wrote my answer on your previous message before seeing this one, haha. Glad we agree then :p